aspnetcore/fundamentals/startup.md

```diff
@@ -59,6 +59,34 @@ For features that require substantial setup, there are `Add[Service]` extension
 
 [!code-csharp[](../common/samples/WebApplication1/Startup.cs?highlight=4,7,11&start=40&end=55)]
 
+::: moniker range=">= aspnetcore-2.1"
+
+<a name="setcompatibilityversion"></a>
+### SetCompatibilityVersion
+
+The `SetCompatibilityVersion` method allows an app to opt-in or opt-out of breaking changes in ASP.NET Core APIs. By opting in, you get the latest behavior, and the long term behavior of ASP.NET Core.
+
+The following code sets the compatibility mode to ASP.NET Core 2.1:
+
+[!code-csharp[Main](startup/sampleCompatibility/Startup.cs?name=snippet1)]
+
+Applications that call `SetCompatibilityVersion(CompatibilityVersion.Version_2_0)` are protected from potentially breaking changes introduced in ASP.NET Core 2.1 and later 2.x versions . This protection does not extend to the next major version.
```

Comment:
I would encourage folks to read the in-code docs about what the switches do (on the options types) and try to use the latest version if it doesn't break them. Most users will not care and will want to opt-in to the latest..

The individual switches have a good (in my opinion) explanation of what changed and why we think it's valuable to our users .

There's more description I've written here as well: https://github.com/aspnet/Mvc/blob/747420e5aa7cc2c7834cfb9731510286ded6fc03/src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/CompatibilitySwitch.cs#L9